### ISSUE TITLE
borgbackup cronjobs not running on omv 5

### ISSUE THREAD

**Issue by tlo8640** [OpenMediaVault-Plugin-Developers/openmediavault-borgbackup#8]
Hello,
I I configured the borgbackup plugin on omv5(beta)
Starting borgbackup from webinterface, everything seems to be fine.
But the cron jobs throws errors. I get the following mails from my omv5 box:

```
Fri Jan 17 10:05:01 CET 2020 Starting backup

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2

Fri Jan 17 10:05:01 CET 2020 Pruning repository

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2

Fri Jan 17 10:05:01 CET 2020 Starting backup

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2

Fri Jan 17 10:05:02 CET 2020 Pruning repository

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2

Fri Jan 17 10:05:02 CET 2020 Starting backup

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2

Fri Jan 17 10:05:02 CET 2020 Pruning repository

Repository /root/{{ ns.uri }} does not exist.
terminating with error status, rc 2
```
Looking into the scripts, there is a difference between the omv 4 script and the omv 5.
im omv 4, the variable BORG_REPO is filled with the right repo name:
```
root@omv:~# cat /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-02e22c68-6ee5-4176-981e-89ab5df63ec3 
#!/bin/sh

# Setting this, so the repo does not need to be given on the commandline:
export BORG_REPO='borgbackup@backupserver:reponame'

```
But in omv5 there is a variable wich seems to be not set during configuration in omv webinterface:
```
root@omv5:/etc/cron.hourly# cat /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d
#!/bin/sh

# This file is auto-generated by openmediavault (https://www.openmediavault.org)
# WARNING: Do not edit this file, your changes will get lost.

# Setting this, so the repo does not need to be given on the commandline:
export BORG_REPO='{{ ns.uri }}'

```

**Comment by ryecoaaron** [comment]
I have this running on my OMV 5.x system and it isn't having this issue.  What is the output of:
dpkg -l | grep borg
sudo omv-salt deploy run borgbackup

**Comment by tlo8640** [comment]
Hello,

thank you for your answer:

Here comes the output from the commands you requested:
First:
```
root@omv5:~# dpkg -l | grep borg
ii  borgbackup                           1.1.9-2                             amd64        deduplicating and compressing backup program
ii  openmediavault-borgbackup            5.0.2                               all          borgbackup plugin for OpenMediaVault.
```
Second:
```
root@omv5:~# omv-salt deploy run borgbackup
debian:
----------
          ID: configure_borg_hourly_dir
    Function: file.directory
        Name: /var/lib/openmediavault/borgbackup/hourly.d
      Result: True
     Comment: Files cleaned from directory /var/lib/openmediavault/borgbackup/hourly.d
     Started: 14:38:20.784320
    Duration: 3.697 ms
     Changes:   
              ----------
              /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d:
                  ----------
                  removed:
                      Removed due to clean
              /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-75a4906f-337f-44f6-9e33-77e31eae20f0:
                  ----------
                  removed:
                      Removed due to clean
              /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-be00c7ff-7ba2-4080-899b-bc2131c5ead4:
                  ----------
                  removed:
                      Removed due to clean
              removed:
                  - /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-75a4906f-337f-44f6-9e33-77e31eae20f0
                  - /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-be00c7ff-7ba2-4080-899b-bc2131c5ead4
                  - /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d
----------
          ID: configure_borg_daily_dir
    Function: file.directory
        Name: /var/lib/openmediavault/borgbackup/daily.d
      Result: True
     Comment: The directory /var/lib/openmediavault/borgbackup/daily.d is in the correct state
     Started: 14:38:20.788091
    Duration: 0.461 ms
     Changes:   
----------
          ID: configure_borg_weekly_dir
    Function: file.directory
        Name: /var/lib/openmediavault/borgbackup/weekly.d
      Result: True
     Comment: The directory /var/lib/openmediavault/borgbackup/weekly.d is in the correct state
     Started: 14:38:20.788608
    Duration: 0.459 ms
     Changes:   
----------
          ID: configure_borg_monthly_dir
    Function: file.directory
        Name: /var/lib/openmediavault/borgbackup/monthly.d
      Result: True
     Comment: The directory /var/lib/openmediavault/borgbackup/monthly.d is in the correct state
     Started: 14:38:20.789127
    Duration: 0.424 ms
     Changes:   
----------
          ID: configure_borg_yearly_dir
    Function: file.directory
        Name: /var/lib/openmediavault/borgbackup/yearly.d
      Result: True
     Comment: The directory /var/lib/openmediavault/borgbackup/yearly.d is in the correct state
     Started: 14:38:20.789603
    Duration: 0.423 ms
     Changes:   
----------
          ID: configure_borg_crond
    Function: file.managed
        Name: /etc/cron.d/openmediavault-borgbackup
      Result: True
     Comment: File /etc/cron.d/openmediavault-borgbackup is in the correct state
     Started: 14:38:20.790081
    Duration: 15.778 ms
     Changes:   
----------
          ID: configure_borg_datengrab_cron_file
    Function: file.managed
        Name: /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d
      Result: True
     Comment: File /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d updated
     Started: 14:38:20.805936
    Duration: 1.412 ms
     Changes:   
              ----------
              diff:
                  New file
              mode:
                  0750
----------
          ID: configure_borg_photoshare_cron_file
    Function: file.managed
        Name: /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-be00c7ff-7ba2-4080-899b-bc2131c5ead4
      Result: True
     Comment: File /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-be00c7ff-7ba2-4080-899b-bc2131c5ead4 updated
     Started: 14:38:20.807408
    Duration: 1.247 ms
     Changes:   
              ----------
              diff:
                  New file
              mode:
                  0750
----------
          ID: configure_borg_userhomes_cron_file
    Function: file.managed
        Name: /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-75a4906f-337f-44f6-9e33-77e31eae20f0
      Result: True
     Comment: File /var/lib/openmediavault/borgbackup/hourly.d/borgbackup-75a4906f-337f-44f6-9e33-77e31eae20f0 updated
     Started: 14:38:20.808714
    Duration: 1.229 ms
     Changes:   
              ----------
              diff:
                  New file
              mode:
                  0750

Summary for debian
------------
Succeeded: 9 (changed=4)
Failed:    0
------------
Total states run:     9
Total run time:  25.130 ms
```
Nothing has changed after running these command:
```
root@omv5:~# grep -R BORG_REPO /var/lib/openmediavault/borgbackup/*
/var/lib/openmediavault/borgbackup/hourly.d/borgbackup-2cd25511-2a21-407e-a6f3-97389fe1f96d:export BORG_REPO='{{ ns.uri }}'
/var/lib/openmediavault/borgbackup/hourly.d/borgbackup-be00c7ff-7ba2-4080-899b-bc2131c5ead4:export BORG_REPO='{{ ns.uri }}'
/var/lib/openmediavault/borgbackup/hourly.d/borgbackup-75a4906f-337f-44f6-9e33-77e31eae20f0:export BORG_REPO='{{ ns.uri }}'
root@omv5:~# grep -R BORG_REPO /var/lib/openmediavault/borgbackup/*
```

**Comment by tlo8640** [comment]
It's really srtrange...
passphrase will be replaced correctly.
and in the xml config in /etc/openmediavault the in the borg backupo section, type is set to "remote" and uri is correct.
running from webinterface everything is working well, generating the cron jobs hast an error somewhere/somehow?

**Comment by ryecoaaron** [comment]
I think this only affects remote settings.  are there any other {{ some_string }} entries in the /var/lib/openmediavault/borgbackup/ scripts?

**Comment by ryecoaaron** [comment]
This should be fixed with commit e33e71471ffb4e5b7a4182ef110daa945f0b4049.  5.0.3 is in the repo now.
